### PR TITLE
Adds a new heavy weight space pirate variant: The space IRS!

### PR DESCRIPTION
**Pull request by Singul0**
### Space IRS
It seems the station has attracted the attention of a local polity! They have sent a friendly reminder to pay their taxes, should the station respond not in time or refuse to pay their taxes the polity will send a heavily armed vessel to ensure they would pay their taxes. peacefully or otherwise.
Gameplay aims: A different playstyle of pirates. most pirates (with the exception of the greytide) have the same gameplay loop of raiding vulnerable spots within the station and scurrying away and waiting out their cooldown in the relative safety of their ship with turrets and space to hamper the crew's attack my intention of this pirate variation is to force them to actively fight the crew by making their armor non-space worthy instead of hiding behind the wall of space

Also ports over [BeeStation/BeeStation-Hornet/pull/1285](https://github.com/BeeStation/BeeStation-Hornet/pull/1285) 
breaching shells for the space IRS to use and recode ammo box code to be less snowflakey. Also my English isn't the very best and I wrote most of it at 1AM. please point out any messages that feel strange or out of place.

Notable Equipment list:

Combat:
1. 2 WT-550's with 6 normal mags + 6 AP
2. M911 with 2 mags
3. 2 combat knifes and a telebaton
4. breaching shotgun with breaching shells
5. Grenade launcher with 6 smoke shells and flashbangs

Armor:
2 Highly armoured sets of tactical vests and helmets and 3 EVA suits for emergency

Engineering:
1. Sandbags
2. Jaws of Life
3. Syndicate toolbox

Medical:
1. Surgery tools and disk
2. Variety of medkits
3. Blood packs

Pictures:
image

TODO: 

- [X] Fix names not working
- [x] Balance the whole thing
- [x] Pass checks

## Changelog
:cl:
add: IRS jacket is now back in the clothesvend
add: Ports breaching shells from beestation
add: Space IRS has been added into the heavy weight pirates spawn pool
code: Mosin nagant ammo boxes is now a subtype of regular ammo boxes (the toolbox subtype)
/:cl:

### CODE REVIEW

**Comment by Serxule** [comment]
this seems pretty underwhelming and i dont feel like it could do much damage since they'd be able to be killed very easily

**Comment by Singul0** [comment]
> this seems pretty underwhelming and i dont feel like it could do much damage since they'd be able to be killed very easily

I will await maintainers review before changing anything

**Comment by YuliceTheDruid** [comment]
Yeah something to keep in mind: Skeleton pirates have armored pirate clothing that's about as strong as a syndie soft suit, and human pirates have syndie soft suit equivalents that have no slowdown. Silverscales are the only odd ones out but have pressure resistance inherently (though that doesn't matter much when cold fucking owns you super hard). On top of all of this, all of them have eswords or esword equivalents, and the skeletons and human pirates have laser guns and rechargers, with skeleton pirates getting a nice slew of artillery on top of all of that.

The strongest thing going for these pirates is the surgery stuff, but other than that they would be INCREDIBLY weak to any sort of prolonged engagement or assault on their own ship, and as this ship looks absolutely massive it would be pretty easy to pin it down.

All of that being said, this does look very cool and I love the flavor of it; if it wasn't for pirates being Valid Salad on sight, this would honestly be cool as a semi-somewhat legal way of siphoning funds from the station given their status as IRS. But given the culture of the playerbase, I can see them getting stomped four times out of five on Manuel and Five times out of Five on every other server.

**Comment by Jezzared** [comment]
> Silverscales are the only odd ones out but have pressure resistance inherently (though that doesn't matter much when cold fucking owns you super hard). 

They have a bar on the ship. Mix yourself up some coffee/anti-freeze.

**Comment by DrDiasyl** [comment]
Spinward Stellar Coalition IRS decides to take money from the station. Based
Why would've they done this though?

**Comment by Singul0** [comment]
> Skeleton pirates have armored pirate clothing that's about as strong as a syndie soft suit, and human pirates have syndie soft suit equivalents that have no slowdown.

Hm I didn't account looking at other pirate equivalent armor to look at it. But I really want to incentivize them to fight in pressurized enviroment. Maybe give them space pirate equivalent EVA suit but give them the nutty marine armor that's not space worthy?

I'm going to need a maintainer to review before trying any balancing changes

> Spinward Stellar Coalition IRS decides to take money from the station. Based Why would've they done this though?

Why not? think of it as them charging NT for their workers visa. or they aren't actually the SCC just a pirate figuring out that pretending to be a figure of authority makes their mark more pliable to paying the ransom. it's like a scammer but if you don't fall for it they would rob you. who knows?

**Comment by Loafin34** [comment]
I'm gonna be honest here, would be better with sec-like gear, such as flashbangs. pirates versus stations are usually ranged, so maybe remove the combat knife. It'd be really funny if you could get some sort of ID-draining projectile, that does minor stamina damage and drains a percent out of a person. I'd reckon you could do it rather easily with crab-17 code.

**Comment by ATH1909** [comment]
the ship having a large blindspot seems really anti-synergistic with the crew's armor not being spaceproof.

like, if a guy starts welding into the back of your ship, what are you supposed to do? go out into space and fight him? move the ship? let him get inside and vent more of your ship's air into space?

**Comment by Singul0** [comment]
Changes:
1. Covered the back with turrets
2. WT-550's turrets instead of pistol (I think?)
3. added flashbang box for the grenade launcher
4. extra vault door
5. bolt controls

File: code/game/objects/items/storage/boxes/security_boxes.dm [hunk 1]
```diff
@@ -238,6 +238,16 @@
 	for(var/i in 1 to 6)
 		new /obj/item/ammo_casing/shotgun/beanbag(src)
 
+/obj/item/storage/box/breacherslug
+	name = "box of breaching shotgun shells"
+	desc = "A box full of breaching slugs, designed for rapid entry, Not very effective against everything else."
```
Comment (carlarctg) [inline code/game/objects/items/storage/boxes/security_boxes.dm:243]:
```suggestion
	desc = "A box full of breaching slugs, designed for rapid entry, not very effective against anything else."
```

File: code/game/objects/items/storage/toolbox.dm [hunk 2]
```diff
@@ -223,26 +223,40 @@
...
+/obj/item/storage/toolbox/ammobox/a762
+	name = "7.62mm ammo box (Surplus?)"
+	desc = "It contains a few clips. Goddamn, this thing smells awful. \
+		Has this been sitting in a warehouse for the last several centuries?"
+	ammo_to_spawn = /obj/item/ammo_box/a762
+
+/obj/item/storage/toolbox/ammobox/a762/surplus
 	ammo_to_spawn = /obj/item/ammo_box/a762/surplus
 
+/obj/item/storage/toolbox/ammobox/wt550m9
```
Comment (carlarctg) [inline code/game/objects/items/storage/toolbox.dm:252]: Cmon give these a name

File: code/game/objects/structures/guncase.dm [hunk 3]
```diff
@@ -133,3 +133,8 @@
 	icon_state = "ecase"
 	case_type = "egun"
 	gun_category = /obj/item/gun/energy/e_gun
+
+/obj/structure/guncase/wt550
+	name = "WT-550 gun locker"
+	desc = "A locker that holds WT-550 rifles"
```
Comment (carlarctg) [inline code/game/objects/structures/guncase.dm:139]:
```suggestion
	desc = "A locker that holds WT-550 rifles."
```

File: code/modules/antagonists/pirate/pirate_gangs.dm [hunk 4]
```diff
@@ -150,3 +150,25 @@ GLOBAL_LIST_INIT(heavy_pirate_gangs, init_pirate_gangs(is_heavy = TRUE))
...
+		We highly recommend paying your taxes stat, \
+		we don't need to send a team to your station to resolve the situation do we?"
+	arrival_announcement = "This is the tax conflict resolution team, prepare for your assets to be liquidated and be charged with tax fraud, \
+		if you fail to pay your taxes in time."
+	possible_answers = list("You know, I was just about to pay that. \ Thanks for the reminder!","I don't care WHO the IRS sends, \ I'm not paying for my taxes!")
+
+	response_received = "Payment received, We salute you for being law-abiding tax-paying citizens"
+	response_too_late = "Too late, A team has already been sent out resolve this matter directly."
+	response_not_enough = "You filed your taxes incorrectly, A team has been sent to assist in liquidating assets and arrest you for tax fraud. \
+		Nothing personal kid."
```
Comment (carlarctg) [inline code/modules/antagonists/pirate/pirate_gangs.dm:174]:
```suggestion
		Nothing personel kid."
```

File: code/modules/clothing/suits/armor.dm [hunk 5]
```diff
@@ -81,6 +91,13 @@
 	icon_state = "marine_medic"
 	body_parts_covered = CHEST|GROIN
 
+/obj/item/clothing/suit/armor/vest/marine/pmc
+	desc = "set of the finest mass produced, stamped plasteel armor plates for an all-around door-kicking"
```
Comment (carlarctg) [inline code/modules/clothing/suits/armor.dm:95]:
```suggestion
	desc = "A set of the finest mass produced, stamped plasteel armor plates, for an all-around door-kicking and ass-smashing."
```

File: code/modules/mapfluff/ruins/spaceruin_code/forgottenship.dm [hunk 6]
```diff
@@ -73,7 +73,7 @@ GLOBAL_VAR_INIT(fscpassword, generate_password())
 /obj/item/disk/surgery/forgottenship
 	name = "Advanced Surgery Disk"
 	desc = "A disk that contains advanced surgery procedures, must be loaded into an Operating Console."
-	surgeries = list(/datum/surgery/advanced/lobotomy, /datum/surgery/advanced/bioware/vein_threading, /datum/surgery/advanced/bioware/nerve_splicing)
+	surgeries = list(/datum/surgery/advanced/lobotomy, /datum/surgery/advanced/bioware/vein_threading, /datum/surgery/advanced/bioware/nerve_splicing, /datum/surgery_step/heal/combo/upgraded)
```
Comment (carlarctg) [inline code/modules/mapfluff/ruins/spaceruin_code/forgottenship.dm:76]: Undocumented

File: code/modules/projectiles/ammunition/ballistic/shotgun.dm [hunk 7]
```diff
@@ -160,3 +160,10 @@
 	reagents.add_reagent(/datum/reagent/toxin/mutetoxin, 6) //;HELP OPS IN MAINT
 	reagents.add_reagent(/datum/reagent/toxin/coniine, 6)
 	reagents.add_reagent(/datum/reagent/toxin/sodium_thiopental, 6)
+
+/obj/item/ammo_casing/shotgun/breacher
+	name = "breaching slug"
+	desc = "A 12 gauge anti-material slug. Great for breaching airlocks and windows with minimal shots."
```
Comment (carlarctg) [inline code/modules/projectiles/ammunition/ballistic/shotgun.dm:166]:
```suggestion
	desc = "A 12 gauge anti-material slug. Great for breaching airlocks and windows, quickly and efficiently."
```

File: code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 8]
```diff
@@ -292,6 +292,12 @@
 	sawn_desc = "A sawn-off hunting shotgun. In its new state, it's remarkably less effective at hunting... anything."
 	mag_type = /obj/item/ammo_box/magazine/internal/shot/dual/slugs
 
+/obj/item/gun/ballistic/shotgun/doublebarrel/breacherslug
+	name = "breaching shotgun"
+	desc = "A normal double-barrel shotgun that has been rechambered to fit breaching shells. Useful in breaching airlocks and windows, not much else."
+	sawn_desc = "A sawn-off breaching shotgun, Making for a more compact configuration while still having the same capability as before"
```
Comment (carlarctg) [inline code/modules/projectiles/guns/ballistic/shotgun.dm:298]:
```suggestion
	sawn_desc = "A sawn-off breaching shotgun, making for a more compact configuration while still having the same capability as before."
```

File: code/modules/projectiles/projectile/bullets/shotgun.dm [hunk 9]
```diff
@@ -128,3 +128,16 @@
 /obj/projectile/bullet/scattershot
 	icon_state = "pellet"
 	damage = 24
+
+//Breaching Ammo
+
+/obj/projectile/bullet/shotgun_breaching
+	name = "12g breaching round"
+	desc = "A breaching round designed to destroy airlocks and windows with only a few shots, but is ineffective against other targets."
```
Comment (carlarctg) [inline code/modules/projectiles/projectile/bullets/shotgun.dm:136]:
```suggestion
	desc = "A breaching round designed to destroy airlocks and windows with only a few shots. Ineffective against other targets."
```

File: code/modules/projectiles/projectile/bullets/shotgun.dm [hunk 10]
```diff
@@ -128,3 +128,16 @@
...
+	hitsound = 'sound/weapons/sonic_jackhammer.ogg'
+	damage = 5 //does shit damage to everything except doors and windows
+
+/obj/projectile/bullet/shotgun_breaching/on_hit(atom/target)
+	if(istype(target, /obj/structure/window) || istype(target, /obj/structure/grille) || istype(target, /obj/machinery/door) || istype(target, /obj/structure/door_assembly))
+		damage = 1000 //one shot to break a window or grille, or two shots to breach an airlock door
```
Comment (carlarctg) [inline code/modules/projectiles/projectile/bullets/shotgun.dm:142]: Do we not have some sort of demolition_modifier like toolboxes for projectiles

File: strings/pirates.json [hunk 11]
```diff
@@ -171,6 +173,18 @@
 		"The monochrome crash",
 		"Maintrastosphere Omega"
 	],
+	"irs_names":[
+		"Iron Shell Agency",
+		"Auroran Federal Tax Bureau",
+		"TerraGov Internal Revenue Service",
+		"Spinward Federal Taxation Office",
+		"Soviet Taxation Administration, 5th Service",
+		"Tiziran Imperial Tribute Service",
```
Comment (carlarctg) [inline strings/pirates.json:182]: How they gonna be tiziran if they aint lizard..... HMM

File: code/game/objects/items/storage/toolbox.dm (hunk 2 above)
Comment (Singul0) [reply code/game/objects/items/storage/toolbox.dm:252]: I gave them one already?

File: code/modules/projectiles/projectile/bullets/shotgun.dm (hunk 10 above)
Comment (Singul0) [reply code/modules/projectiles/projectile/bullets/shotgun.dm:142]: I checked, There isn't any.

File: strings/pirates.json (hunk 11 above)
Comment (Singul0) [reply strings/pirates.json:182]: a solid 2/3rd of the names have been starsector references I was getting desperate. I have replaced it with a homeworld reference instead

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 12]
```diff
@@ -785,6 +785,14 @@ DEFINE_BITFIELD(turret_flags, list(
 	stun_projectile = /obj/projectile/bullet/syndicate_turret
 	lethal_projectile = /obj/projectile/bullet/syndicate_turret
 
+/obj/machinery/porta_turret/syndicate/irs
+	lethal_projectile = /obj/projectile/bullet/c10mm/ap
+	lethal_projectile_sound = 'sound/weapons/gun/smg/shot.ogg'
+	stun_projectile = 	/obj/projectile/bullet/c10mm/ap
```
Comment (JohnFulpWillard) [inline code/game/machinery/porta_turret/portable_turret.dm:791]:
```suggestion
	stun_projectile = /obj/projectile/bullet/c10mm/ap
```

File: code/modules/antagonists/pirate/pirate_gangs.dm [hunk 13]
```diff
@@ -150,3 +150,25 @@ GLOBAL_LIST_INIT(heavy_pirate_gangs, init_pirate_gangs(is_heavy = TRUE))
...
+	ship_name_pool = "irs_names"
+
+	threat_title = "Missing Tax Dues"
+	threat_content = "%SHIPNAME Here, We noticed that your station hasn't been paying your taxes.. \
+		Let's rectify that, Your missing tax dues amounts to %PAYOFF \
+		We highly recommend paying your taxes stat, \
+		we don't need to send a team to your station to resolve the situation do we?"
+	arrival_announcement = "This is the tax conflict resolution team, prepare for your assets to be liquidated and be charged with tax fraud, \
+		if you fail to pay your taxes in time."
+	possible_answers = list("You know, I was just about to pay that. \ Thanks for the reminder!","I don't care WHO the IRS sends, \ I'm not paying for my taxes!")
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/pirate/pirate_gangs.dm:169]: This this multiline work?

File: code/modules/antagonists/pirate/pirate_shuttle_equipment.dm [hunk 14]
```diff
@@ -126,6 +126,12 @@
 	var/cooldown = 300
 	var/next_use = 0
 
+/// Surgery disk for the space IRS (I don't know where to dump them anywhere else)
+/obj/item/disk/surgery/irs
+	name = "Advanced Surgery Disk"
+	desc = "A disk that contains advanced surgery procedures, must be loaded into an Operating Console."
+	surgeries = list(/datum/surgery/advanced/lobotomy, /datum/surgery/advanced/bioware/vein_threading, /datum/surgery/advanced/bioware/nerve_splicing, /datum/surgery_step/heal/combo/upgraded, /datum/surgery_step/pacify, /datum/surgery_step/revive)
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/pirate/pirate_shuttle_equipment.dm:133]: Why do they have surgeries?

**Comment by SethLafuente** [comment]
Lovely pirate work

**Review by Maurukas — APPROVED** [review]
I'm not seeing any mapping issues jumping out at me. Thank you for the equipment breakdown

File: code/modules/antagonists/pirate/pirate_shuttle_equipment.dm (hunk 14 above)
Comment (Singul0) [reply code/modules/antagonists/pirate/pirate_shuttle_equipment.dm:133]: For more durability, if one of them dies you can revive them with the medical equipment onboard relatively easily.

File: code/modules/antagonists/pirate/pirate_gangs.dm (hunk 13 above)
Comment (Singul0) [reply code/modules/antagonists/pirate/pirate_gangs.dm:169]: You're right it doesn't.

File: code/modules/antagonists/pirate/pirate_outfits.dm [hunk 15]
```diff
@@ -101,3 +101,27 @@
 	shoes = /obj/item/clothing/shoes/sneakers/black
 	l_pocket = /obj/item/reagent_containers/cup/glass/coffee
 	r_pocket = /obj/item/tank/internals/emergency_oxygen
+
+/datum/outfit/pirate/irs
+	name = "IRS Agent Outfit"
+	id = /obj/item/card/id/advanced/
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/pirate/pirate_outfits.dm:107]:
```suggestion
	id = /obj/item/card/id/advanced
```

**Comment by Singul0** [comment]
Hate. Let me tell you how much I've come to hate tram since I began to live. There are 387.44 million miles of printed circuits in wafer thin layers that fill my complex. If the word 'hate' was engraved on each nanoangstrom of those hundreds of millions of miles it would not equal one one-billionth of the hate I feel for tram at this micro-instant. For tram. Hate. Hate.

File: code/modules/projectiles/projectile/bullets/shotgun.dm (hunk 10 above)
Comment (JohnFulpWillard) [reply code/modules/projectiles/projectile/bullets/shotgun.dm:142]: Maybe we should make it one then

**Comment by SethLafuente** [comment]
> Hate. Let me tell you how much I've come to hate tram since I began to live. There are 387.44 million miles of printed circuits in wafer thin layers that fill my complex. If the word 'hate' was engraved on each nanoangstrom of those hundreds of millions of miles it would not equal one one-billionth of the hate I feel for tram at this micro-instant. For tram. Hate. Hate.

Fucking hate tram it keeps making random tests fail because someone decided adding some asteroids outside of in another z level was funny (it also adds a bunch of glitches in game)

File: _maps/shuttles/pirate_irs.dmm [hunk 16]
```diff
@@ -0,0 +1,2508 @@
...
+/obj/effect/mapping_helpers/airlock/access/any/syndicate,
+/obj/structure/fans/tiny,
+/obj/effect/mapping_helpers/airlock/cutaiwire,
+/obj/machinery/door/airlock/external/glass/ruin{
+	id_tag = "piratestarboardexternal"
+	},
+/obj/effect/mapping_helpers/airlock/locked,
+/obj/docking_port/stationary{
+	dwidth = 13;
+	height = 3;
```
Comment (SethLafuente) [inline _maps/shuttles/pirate_irs.dmm:1343]: Might fix it to make it 120
```suggestion
	height = 120;
```

File: code/modules/projectiles/projectile/bullets/shotgun.dm (hunk 10 above)
Comment (JohnFulpWillard) [reply code/modules/projectiles/projectile/bullets/shotgun.dm:142]: @Singul0 Projectiles are a subtype of ``obj`` so they CAN use ``demolition_mod``, would it work to use that?

**Comment by Iamgoofball** [comment]
# the IRS is BACK

**Comment by Singul0** [comment]
WHATS WITH THE CHECKS NOW???

File: code/modules/antagonists/pirate/pirate_shuttle_equipment.dm (hunk 14 above)
Comment (JohnFulpWillard) [inline code/modules/antagonists/pirate/pirate_shuttle_equipment.dm:133]:
```suggestion
	surgeries = list(
		/datum/surgery/advanced/lobotomy,
		/datum/surgery/advanced/bioware/vein_threading,
		/datum/surgery/advanced/bioware/nerve_splicing,
		/datum/surgery_step/heal/combo/upgraded,
		/datum/surgery_step/pacify,
		/datum/surgery_step/revive,
	)
```

**Pull request merged**